pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -0,0 +1,173 @@
+                        { _("Close") }
+                    </Button>
+                </Modal.Footer>
+            </Modal>
+        );
+    }
+}
+
+ModificationsExportDialog.propTypes = {
+    shell: PropTypes.string.isRequired,
+    ansible: PropTypes.string,
+    show: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+};
+
+/* Display list of modifications in human readable format
+ *
+ * Also show `View automation script` button which opens dialog in which different
+ * scripts are available. With these scripts it is possible to apply the same
+ * configurations to  other machines.
+ *
+ * Pass array `entries` to show human readable messages.
+ * Pass string `shell` and `ansible` with scripts.
+ *
+ */
+export class Modifications extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showDialog: false,
+        };
+    }
+
+    render() {
+        let emptyRow = null;
+        if (this.props.entries === null) {
+            emptyRow = <thead className={"listing-ct-empty"}>
+                <tr className="modification-row">
+                    <td>
+                        <div className="spinner spinner-sm" />
+                        <span>{_("Loading system modifications...")}</span>
+                    </td>
+                </tr>
+            </thead>;
+        }
+        if (this.props.entries !== null && this.props.entries.length === 0) {
+            emptyRow = <thead className={"listing-ct-empty"}>
+                <tr className="modification-row">
+                    <td>
+                        { this.props.permitted ? _("No System Modifications") : _("The logged in user is not permitted to view system modifications") }
+                    </td>
+                </tr>
+            </thead>;
+        }
+
+        return (
+            <React.Fragment>
+                <ModificationsExportDialog show={this.state.showDialog} shell={this.props.shell} ansible={this.props.ansible} onClose={ () => this.setState({ showDialog: false }) } />
+                <table className={"listing-ct listing-ct-wide modifications-table"}>
+                    <caption className="cockpit-caption">
+                        <div className="modifications-caption">
+                            {this.props.title}
+                            {!emptyRow &&
+                                <a className="modifications-export" onClick={ () => this.setState({ showDialog: true }) } >{_("View automation script")}</a>
+                            }
+                        </div>
+                    </caption>
+                    {emptyRow}
+                    { !emptyRow &&
+                        <tbody>
+                            {this.props.entries.map(entry => <tr className="modification-row" key={entry.split(' ').join('')}><td>{entry}</td></tr>)}
+                        </tbody>
+                    }
```

Comment:
This could be written a tad more elegantly with a single `{ emptyRow || <tbody>...}`, or maybe calling the variable `row` and assigning one or the other to it? Then this would also make this a bit more readable.